projects/main/src/app/models/cosmos/bank.application.service.ts

```diff
@@ -35,11 +36,29 @@ export class BankApplicationService {
 
     const dialogRefSimulating = this.loadingDialog.open('Simulating...');
 
+    //Todo: if amount has maximum gasDenom, do amount -1
+    const maximumGasPrice = coins.find((coin) => coin.denom === minimumGasPrice.denom);
+    const amountGasPrice = amount.find((amount) => amount.denom === minimumGasPrice.denom);
+
+    const fixedAmount: proto.cosmos.base.v1beta1.ICoin[] = [];
+    amount.forEach((amount) => {
+      //has gas denom
+      if (
+        maximumGasPrice?.amount === amountGasPrice?.amount &&
+        amount.denom === minimumGasPrice.denom
+      ) {
+        const amountForSimulation = parseInt(amount.amount || '0') - 1;
+        fixedAmount.push({ amount: amountForSimulation.toString(), denom: amount.denom });
+      } else {
+        fixedAmount.push(amount);
+      }
+    });
```

Comment:
間違っているわけではなく好みの問題かもしれませんが、空の配列を新たに定義してpushで詰めていくより、元となる配列からmapで生成した結果として一発で定義したほうが自然かなと思いました。

```suggestion
    const fixedAmount: proto.cosmos.base.v1beta1.ICoin[] = amount.map((amount) => {
      //has gas denom
      if (
        maximumGasPrice?.amount === amountGasPrice?.amount &&
        amount.denom === minimumGasPrice.denom
      ) {
        const amountForSimulation = parseInt(amount.amount || '0') - 1;
        return { amount: amountForSimulation.toString(), denom: amount.denom };
      } else {
        return amount;
      }
    });
```